invisible_cities/cities/esmeralda.py

```diff
@@ -164,96 +138,102 @@ class HitEnergy(AutoNameEnumBase):
-            extr2_pos = extr2.XYZ
-
-            blob_pos1, blob_pos2 = plf.blob_centres(t, blob_radius)
-
-            e_blob1, e_blob2, hits_blob1, hits_blob2 = plf.blob_energies_and_hits(t, blob_radius)
-            overlap = False
-            overlap = sum([h.Ec for h in set(hits_blob1).intersection(hits_blob2)])
-            list_of_vars = [hitc.event, tID, energy, length, numb_of_voxels,
-                            numb_of_hits, numb_of_tracks,
-                            min_x, min_y, min_z, max_x, max_y, max_z, max_r,
-                            ave_pos[0], ave_pos[1], ave_pos[2],
-                            extr1_pos[0], extr1_pos[1], extr1_pos[2],
-                            extr2_pos[0], extr2_pos[1], extr2_pos[2],
-                            blob_pos1[0], blob_pos1[1], blob_pos1[2],
-                            blob_pos2[0], blob_pos2[1], blob_pos2[2],
-                            e_blob1, e_blob2, overlap,
-                            vox_size_x, vox_size_y, vox_size_z]
-
-            df.loc[c] = list_of_vars
-            try:
-                types_dict
-            except NameError:
-                types_dict = dict(zip(df.columns, [type(x) for x in list_of_vars]))
-
-            for vox in t.nodes():
-                for hit in vox.hits:
-                    hit.track_id = tID
-                    track_hits.append(hit)
-
-
-        track_hitc = evm.HitCollection(hitc.event, hitc.time)
-        track_hitc.hits = track_hits
-        #change dtype of columns to match type of variables
-        df = df.apply(lambda x : x.astype(types_dict[x.name]))
 
-        return df, track_hitc
+        if len(hitc.hits)>0:
+            voxels     = plf.voxelize_hits(hitc.hits, vox_size, strict_vox_size, evm.HitEnergy.Ec)
+            mod_voxels = plf.drop_end_point_voxels(voxels, energy_threshold, min_voxels)
+            tracks     = plf.make_track_graphs(mod_voxels)
+
+            vox_size_x = voxels[0].size[0]
+            vox_size_y = voxels[0].size[1]
+            vox_size_z = voxels[0].size[2]
+            del(voxels)
+            #sort tracks in energy
+            tracks     = sorted(tracks, key=plf.get_track_energy, reverse=True)
+
+            track_hits = []
+            for c, t in enumerate(tracks, 0):
+                tID = c
+                energy = plf.get_track_energy(t)
+                length = plf.length(t)
+                numb_of_hits   = len([h for vox in t.nodes() for h in vox.hits])
+                numb_of_voxels = len(t.nodes())
+                numb_of_tracks = len(tracks   )
+
+                pos   = [h.pos for v in t.nodes() for h in v.hits]
+                min_x = min(pos, key = lambda x : x[0])[0]
+                max_x = max(pos, key = lambda x : x[0])[0]
+                min_y = min(pos, key = lambda x : x[1])[1]
+                max_y = max(pos, key = lambda x : x[1])[1]
+                min_z = min(pos, key = lambda x : x[2])[2]
+                max_z = max(pos, key = lambda x : x[2])[2]
```

Comment:
mmm, I was thinking about something in the lines of
```
pos = bla
x, y, z = map(np.array, zip(*pos))
r = np.sqrt(x**2 + y**2)
```
and then mins and maxes are trivial.